### PR TITLE
Update Customer.io destination with new mapping

### PR DESCRIPTION
**Pull request by sydneycollins-cio**
Updates the Customer.io destination:

Updated destinations

`createUpdateObject`
`deleteDevice`
`deleteObject`
`deleteRelationship`

## Testing

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [ ] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [ ] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [ ] [Segmenters] Tested in the staging environment

### CODE REVIEW

File: packages/destination-actions/src/destinations/customerio/deleteObject/index.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ const action: ActionDefinition<Settings, Payload> = {
       description: 'An object ID used to identify an object.',
       type: 'string',
       default: {
-        '@path': '$.context.groupId'
+        '@path': '$.groupId'
```
Comment (joe-ayoub-segment) [inline packages/destination-actions/src/destinations/customerio/deleteObject/index.ts:16]: Hi @sydneycollins-cio - would it not make sense to do this following?

```suggestion
        '@path': '$.properties.objectId'
```

File: packages/destination-actions/src/destinations/customerio/createUpdatePerson/index.ts [hunk 2]
```diff
@@ -118,6 +118,8 @@ function mapPayload(payload: Payload) {
   delete custom_attributes.created_at
   delete custom_attributes?.object_type_id
   delete custom_attributes?.relationshipAttributes
+  delete (custom_attributes as { traits?: { object_type_id?: unknown } })?.traits?.object_type_id
+  delete (custom_attributes as { traits?: { relationshipAttributes?: unknown } })?.traits?.relationshipAttributes
```
Comment (joe-ayoub-segment) [inline packages/destination-actions/src/destinations/customerio/createUpdatePerson/index.ts:122]: hi @sydneycollins-cio - can you explain what's supposed to be happening here, and how it differs from these lines ? 
```
delete custom_attributes?.object_type_id
 delete custom_attributes?.relationshipAttributes
```

File: packages/destination-actions/src/destinations/customerio/createUpdateObject/index.ts [hunk 3]
```diff
@@ -145,6 +145,8 @@ function mapPayload(payload: Payload) {
     object_id: id
   }
 
+  delete (custom_attributes as { traits?: { object_type_id?: unknown } })?.traits?.object_type_id
+  delete (custom_attributes as { traits?: { relationshipAttributes?: unknown } })?.traits?.relationshipAttributes
```
Comment (joe-ayoub-segment) [inline packages/destination-actions/src/destinations/customerio/createUpdateObject/index.ts:149]: hi @sydneycollins-cio can you explain what these lines are attempting to do please?

Comment (joe-ayoub-segment) [reply packages/destination-actions/src/destinations/customerio/createUpdateObject/index.ts:149]: Hi @sydneycollins-cio thanks for clarifying. 

I'm not sure this is going to work, however the code shouldn't cause any issues so it is OK to deploy. 

`delete (custom_attributes as { traits?: { object_type_id?: unknown } })?.traits?.object_type_id` 
will attempt to delete this attribute: 
`custom_attributes.traits.object_type_id`

However I can't see how custom_attributes will ever contain the traits object. 

Either way, I'm ok to deploy this - you can amend / fix if it doesn't work as you need.

**Review by joe-ayoub-segment — APPROVED** [review]

**Review by joe-ayoub-segment — DISMISSED** [review]

**Review by joe-ayoub-segment — APPROVED** [review]

**Comment by joe-ayoub-segment** [comment]
hi @sydneycollins-cio this PR has been deployed. Can you please confirm that you are happy with the change?

**Comment by sydneycollins-cio** [comment]
> hi @sydneycollins-cio this PR has been deployed. Can you please confirm that you are happy with the change?

It looks great! thanks so much 🙏🏽

**Pull request merged**